src/agent/codecoveragesummaryreader.ts

```diff
@@ -0,0 +1,219 @@
+                    coverageStats.push(coverageStat);
+                }
+            }
+            coverage.addResults(coverageStats);
+        }
+        catch (error) {
+            defer.reject(error);
+            return defer.promise;
+        }
+
+        defer.resolve(coverage);
+        return defer.promise;
+    }
+}
+
+export class CoberturaSummaryReader implements ccp.ICodeCoverageReader {
+
+    public command: cm.ITaskCommand;
+
+    constructor(command: cm.ITaskCommand) {
+        this.command = command;
+    }
+
+    public getCodeCoverageSummary(summaryFilePath: string): Q.Promise<testifm.CodeCoverageData> {
+        var defer = Q.defer<testifm.CodeCoverageData>();
+        var _this = this;
+
+        SummaryReaderUtilities.getXmlContent(summaryFilePath).then(function(xmlContents) {
+            _this.parseCoberturaXmlReport(xmlContents).then(function(codeCoverageSummary) {
+                defer.resolve(SummaryReaderUtilities.getCodeCoverageData(codeCoverageSummary));
+            }).fail(function(err) {
+                defer.reject(err);
+            });
+        }).fail(function(err) {
+            defer.reject(err);
+        });
+
+        return defer.promise;
+    }
+
+    private parseCoberturaXmlReport(xmlContent): Q.Promise<CodeCoverageSummary> {
+
+        this.command.info("parseCoberturaXmlReport: Parsing summary file.");
+        var defer = Q.defer<CodeCoverageSummary>();
+
+        if (!xmlContent.coverage || !xmlContent.coverage.at(0) || !xmlContent.coverage.at(0).attributes()) {
+            defer.resolve(null);
+        }
+
+        var coverageStats = [];
+        var coverage = new CodeCoverageSummary();
+        try {
+            var coverageNode = xmlContent.coverage.at(0);
+            var attributes = coverageNode.attributes();
+
+            var linesTotal = attributes['lines-valid'];
+            var linesCovered = attributes['lines-covered'];
+            var branchesCovered = attributes['branches-covered'];
+            var branchesTotal = attributes['branches-valid'];
+
+            if (linesTotal && linesCovered) {
+                this.command.info("Lines : " + linesCovered + "/" + linesTotal + " covered.");
+                var coverageStat = SummaryReaderUtilities.getCodeCoverageStatistics("Lines", linesCovered, linesTotal, "line");
+                coverageStats.push(coverageStat);
+            }
+
+            if (branchesCovered && branchesTotal) {
+                this.command.info("Branches : " + branchesCovered + "/" + branchesTotal + " covered.");
+                var coverageStat = SummaryReaderUtilities.getCodeCoverageStatistics("Branches", branchesCovered, branchesTotal, "branch");
+                coverageStats.push(coverageStat);
+            }
+
+            coverage.addResults(coverageStats);
+        }
+        catch (error) {
+            defer.reject(error);
```

Comment:
I have not logged here coz they will be logged in the codecoverage publisher calls. didt want duplicate logs